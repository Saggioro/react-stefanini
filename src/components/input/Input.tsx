import React, { FunctionComponent } from "react";
import { Container, Label, InputTag, Error } from "./Input.styled";

interface IInput extends React.HTMLAttributes<HTMLInputElement> {
  width?: number;
  height?: number;
  label?: string;
  value: string;
  name: string;
  error?: string;
  type?: string;
}

const Input: FunctionComponent<IInput> = ({
  width,
  label,
  height,
  value,
  name,
  error,
  type,
  ...props
}: IInput) => {
  return (
    <Container>
      <Label>{label}</Label>
      <InputTag
        type={type || "text"}
        name={name}
        value={value}
        height={height}
        width={width}
        {...props}
      />
      <Error>{error}</Error>
    </Container>
  );
};

export default Input;
