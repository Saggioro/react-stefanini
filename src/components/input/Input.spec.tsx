/* eslint-disable testing-library/prefer-screen-queries */
/* eslint-disable testing-library/no-debugging-utils */
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import Input from "./Input";
jest.mock("react-redux", () => {
  return {
    useDispatch() {
      return {
        dispatch: (user: any) => {
          user = undefined;
        },
      };
    },
  };
});

test("Input should render correctly", () => {
  const component = renderer.create(
    <Input name="test" value="test" onChange={() => {}} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("Input label should render", () => {
  render(<Input name="test" value="test" label="label" onChange={() => {}} />);

  expect(screen.getByText("label")).toBeInTheDocument();
});
