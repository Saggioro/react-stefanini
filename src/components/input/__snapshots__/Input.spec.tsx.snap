// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<div
  className="sc-bczRLJ goEvZo"
>
  <label
    className="sc-dkzDqf hbcVSe"
  />
  <input
    className="sc-gsnTZi flIaCB"
    name="test"
    onChange={[Function]}
    type="text"
    value="test"
  />
  <span
    className="sc-hKMtZM ftuqSP"
  />
</div>
`;
